lib/consensus.js

```diff
@@ -974,10 +974,24 @@ function _tallyBranches({
           _computeMostRecentVotes({event, yElectorSet, yByElector});
         }
         // if some voting ancestors haven't computed support yet, defer
-        const votingEvents = Object.values(event._votes);
-        if(votingEvents.some(
-          e => e && e !== event && e._supporting === undefined)) {
-          next.push(event);
+        let pushed = false;
+        for(const v in event._votes) {
+          const e = event._votes[v];
+          // FIXME: remove this conditional, e is undefined in exactly one
+          // place in the test
+          // suite which is some of the findConsensus figures. Specifically
+          // seems to be in relation to a node `b` there.
+          if(!e) {
```

Comment:
> That said, we never actually fall into the conditional on the new line 988 except in synthetic ledger histories in the test suite. So I would like to discuss the necessity for some of this code existing in the first place.

`e` is false when a byzantine node (fork) is detected, so this is vital. It only happens in "synthetic ledger histories" as you say, or rather, in the test vectors, because it is for the correctness testing for handling byzantine nodes. Our software should never ("can" never) misbehave in this way (fork), so we test this behavior via those test vectors.